internal/hcpvalidator/url.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package hcpvalidator
+
+import (
+	"context"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+const (
+	invalidURLErr = "must be a valid URL"
+)
+
+var (
+	_ validator.String = urlValidator{}
+)
+
+// urlValidator validates that a string Attribute's value is a valid URL.
+type urlValidator struct {
```

Comment:
Seems like you could store a boolean here `httpsOnly bool` and then use the same validator for both url and HTTPSUrl.

With a separate constructor:

```
func HTTPSURL() validator.String {
   return urlValidator{
      httpsOnly: true,
    }
 }
```